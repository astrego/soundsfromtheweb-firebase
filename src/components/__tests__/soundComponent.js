import React from 'react';
import renderer from 'react-test-renderer';
import SoundComponent from '../soundComponent';
import * as Gatsby from 'gatsby';

describe('testComponent', () => {
  const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
  useStaticQuery.mockImplementation(() => ({
    file: {
      publicURL: '/static/dfe0312a9e62d182bd5c2a320ae37016/testopname.WAV',
    },
  }));

  it('renders correctly', () => {
    const tree = renderer.create(<SoundComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
