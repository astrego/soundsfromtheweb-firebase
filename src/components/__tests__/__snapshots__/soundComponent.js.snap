// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testComponent renders correctly 1`] = `
<div>
  <h1>
    Sound Component
  </h1>
  <audio
    controls={true}
  >
    <source
      src="test-file-stub"
      type="audio/mpeg"
    />
    Your browser does not support the audio element.
  </audio>
  <br />
  <br />
  <div>
    <a
      download={true}
      href="/static/dfe0312a9e62d182bd5c2a320ae37016/testopname.WAV"
      style={
        Object {
          "backgroundColor": "blue",
          "borderRadius": 5,
          "color": "white",
          "padding": 12,
          "textDecoration": "none",
        }
      }
    >
      Download the high quality WAV
    </a>
  </div>
</div>
`;
